<blockquote>
<h1>Environmental Epigenetics</h1>
</blockquote>

##### Developing genomic resources to support restoration and protection of the Olympia Oyster in Puget Sound
_Summary_: There is a significant gap in our fundamental understanding of this species’ resilience in the face of environmental change, ecological interactions, and population structure. This information is critical to local restoration efforts and to predicting how molluscs will adapt to long-term environmental change. There is recent evidence that oysters have the capacity to respond to environmental change at a rate beyond what would be predicted by genetic variation alone. The overall objective of this research is to produce genomic resources and capacity to understand the response of Olympia oysters to environmental change. Specifically, a draft genome assembly for the Olympia oyster will be produced and used to understand how responses of the Olympia oyster to environmental changes are inherited (i.e., genetic or epigenetic) using restriction site associated DNA sequencing (RAD-Seq) and bisulfite sequencing (BS-Seq). A web-based platform will be developed based on these resources that will be used for discovery and further collaboration.

##### Assessing the Capacity for Evolutionary Adaptation to Ocean Acidification in Geoduck
_Summary_: We will investigate the roles of standing genetic (DNA sequence) and epigenetic (DNA methylation) variation in the response to ocean acidification and elevated temperature in geoduck at the vulnerable larval stage. The specific research objectives are to (1) characterize changes in allele frequencies under ocean acidification and temperature stress at single nucleotide polymorphisms (SNPs) throughout the genome, and (2) determine the change in frequency of methylation states (epialleles) under ocean acidification and temperature stress throughout the epigenome. To meet our objectives, we will employ a within-generation selection experiment on larvae subjected to ambient and high pCO2 and ambient and high temperature in a factorial cross. This experimental design allows us to uncover changes in standing genetic and epigenetic variation in response to ocean acidification and elevated temperature, and will provide essential information on how marine invertebrate populations can respond to, and persist in, an acidifying ocean.

##### Evaluating the role of DNA methylation in phenotypic plasticity and response to environmental change in tropical reef corals
_Summary_: Phenotypic plasticity is critical to the survival of many organisms in a rapidly changing environment, especially threatened species like tropical reef corals. This project will assess whether DNA methylation plays a role in coral phenotypic plasticity. Caribbean corals of the genus _Porites_ will be used as primary research models. The project will first assess whether differential methylation patterns are associated with alternative phenotypes, as well as determine whether methylation and phenotype change and co-vary in reponse to experimental manipulations. Further work will evaluate relationships between methylation and transcription, test for a causative role of DNA methylation in coral thermal tolerance plasticity, and assess heritability of DNA methylation by comparison of methylation patterns in adult corals and their larval offspring. 


 
<blockquote>
<h1>Reproductive Biology and Ecology</h1>
</blockquote>

##### Using satellite pop-up tags to track movements of sablefish during spawning and changes in vertical position in the water column
_Summary_: The Sablefish (_Anoplopoma fimbria_) is a deep-water groundfish species widely distributed throughout the northern Pacific Ocean. While some broadscale movements of sablefish have been addressed with tagging studies, two fundamental questions related to sablefish movements are 1) Where do sablefish go during spawning and; 2) Are sablefish adults exclusively benthic inhabitants? To address these questions we will use “pop-up” satellite tag (PSAT) technology. Pop-up satellite tags are composed of a data-logger and battery, and a float with attached antenna. The tag is tethered to the fish and the data logger is programmed to continually collect depth, temperature, and location for a specified duration (weeks to months). At the end of the programmed time, a detachment mechanism is automatically activated that releases the data logger/float and antenna from the tether. The float carries the data logger to the surface where the antenna transmits the data to an orbiting satellite of the ARGOS System that then transmits data back to the researcher. Time series analysis will be employed to determine periodicity in depth profiles and relationships with other environmental parameters such as temperature. Data from this study will allow us to discern daily/weekly and monthly patterns in vertical positioning in the water column and will provide information on possible spawning locations.

##### Eelgrass as refuge: exploring diurnal distribution of bivalve larvae across habitats
_Summary_: To evaluate the distribution of Pacific oyster (_Crassostrea gigas_) and geoduck (_Panopea generosa_) larvae across eelgrass (Zostera marina) and unstructured habitats, and to quantify temporal and spatial variation in larval supply across sites in Puget Sound and the Washington coast through the collection of field data in targeted experiments. Results will assist scientists in developing stewardship recommendations to adaptively manage leased activities and wildstock geoduck harvest on State Owned Aquatic Lands.

##### Proteomic response of shellfish to environmental stress
_Summary_: Shellfish in Puget Sound face a changing environment, as climate change drives increased water temperature and ocean acidification results in decreased pH/Ωaragonite.  Careful laboratory work has shown that shellfish demonstrate a sub-lethal response to these stressors, even when they exhibit no change in rates of survival or growth. This part of the project will measure when and where outplanted _C. gigas_ and _P. generosa_ show a sub-lethal response in protein expression and morphology across sites that span Washington waters. The results will shed light on the relative stress burden of _C. gigas_ and _P. generosa_ populations on SOAL, guiding future work on restoration priorities.



<blockquote>
<h1>Aquaculture Resource Development</h1>
</blockquote>

##### Engaging Students and Public in Marine Conservation through Sustainable Shellfish Aquaculture
_Summary_: The UW Shellfish Farm is a project conceived by several graduate students, faculty and staff at the School of Aquatic &amp; Fishery Sciences (SAFS), in collaboration with the School of Marine &amp; Environmental Affairs (SMEA) and the College of the Environment. We seek to establish a student-run shellfish farm at the Big Beef Creek Research Station, a SAFS field site on Hood Canal. The overarching goals of the project are:
1) Provide students and with a hands-on opportunity to experience sustainable seafood production.
2) Conduct outreach and education on estuarine health and climate change impacts on the marine environment.
3) Conduct monitoring to quantify environmental impacts of shellfish aquaculture on estuarine health.
4) Serve as a field site for ongoing shellfish research at UW.
5) Become self-sustaining by selling cultivated shellfish (e.g. clams, oysters) through wholesalers, UW dining establishments, a subscription service, and other outlets.
[more info](https://csf.uw.edu/project/531)

##### A novel proteomic-based approach to identify and mitigate factors responsible for shellfish mortality events
_Summary_: The overall goal of this project is use a ‘bottom- up’ approach to investigate the process of Pacific oyster larval development. This includes identifying the most susceptible and environmentally sensitive stage of the setting process, characterizing the cause of mass mortalities, and providing guidance and advice for testing and altering culture practices that can obviate seed mortality. The effort includes a discovery-based proteomics approach to reveal critical differences in protein expression profiles between groups of Pacific oysters at three early life stages that are destined to die or survive.  The identified suite of proteins will then be used in a targeted proteomics approach to develop a reproducible, quantifiable assay.  The novel assay will be used to test specific hypotheses based on the response of Pacific oysters to their environment in hatchery and wild settings. An integrated collaboration between researchers, K-12 teachers, and students will build knowledge and increase connections among educators, the shellfish industry, and the scientific community. The specific research objectives of this project are to, 1a) Identify protein biomarkers indicative of Pacific oyster larvae mortality events, 1b) Characterize fundamental physiological changes in Pacific oyster larvae development, 2a) Develop targeted assay to query specific suites of proteins, and 2b) Perform the targeted assay on hatchery produced and wild Pacific oysters.

##### Crossbreeding and Selection for Resistance to Ocean Acidification in Pacific Oysters
_Summary_: We propose to utilize the large number of genetically distinct, pair-mated Pacific oyster lines produced through Taylor Shellfish Company’s breeding program to initiate large-scale and high throughput testing for evaluating differential response in Pacific oysters during two critical and sensitive larval life history phases to increased levels of pCO2 under experimental conditions. Information on relative resilience (measured as survivorship, growth and lipid energy reserves) to OA conditions in genetically distinct oyster lines will inform and guide the production of genetically improved broodstocks and provide the shellfish industry with better surviving/performing intraspecific hybrid oysters. This research will also enable us to greatly expand upon ongoing WA Sea Grant supported work that is investigating the critical importance of OA exposure during the sexual development phase (e.g. conditioning) in adults on subsequent larval and seed survivorship and growth (e.g. intergenerational effects).

#### Characterization of SNPs in genes related to growth from transcriptomes of the flounder _Paralichthys adspersus_
_Summary_: Project: The main goal of this project is to: identify SNPs related to development and growth in larvae and juveniles of the flounder _P. adspersus_ by the comparison of complete transcriptomes. The aims include 1)	Characterize transcriptomes during early stages (larvae) of development, 2)	Evaluate the transcriptome from different tissues of juveniles (of 6 months post hatching) under captivity with normal and low growth, and evaluate the expression patterns of genes associated with the growth of larvae compared to juveniles, and growth patterns, 3) Identify SNPs markers in genes associated with growth and its relationship to gene expression level, and 4) Validation and characterization of SNPs related to growth in different flounder families keep under controlled conditions.


	

